app/Filament/Resources/HubResource.php

```diff
@@ -85,6 +95,12 @@ public static function form(Form $form): Form
                             ->minLength(2)
                             ->maxLength(80)
                             ->rules([new NoEmoji()]),
+
+                        Forms\Components\Toggle::make('has_content_warning')
+                            ->label('Has Content Warning')
+                            ->helperText('CAUTION: If this is enabled, players will get a warning when opening any game in the hub!')
+                            ->default(false)
+                            ->visible(fn ($record) => $user->can('toggleHasContentWarning', $record)),
```

Comment:
I can't decide if this is a poorly implemented feature, or a stub for future functionality.

As it stands, this is specifically for the "mature content warning", but it seems like it could be for any type of content warning (i.e. seizure warnings). I'm assuming that in the the future we want to break "mature content" into more categories, like: language, nudity, sex, abuse. Otherwise, there would be little benefit to allowing the value to be changed by users.

Maybe it should be renamed to `has_mature_content` or something similar. Or changed to an enum that can be set to "mature" or "seizure" or whatever other warnings we want to add in the future.

Reply:
> I can't decide if this is a poorly implemented feature, or a stub for future functionality.

The intention of this feature is to formalize our existing mature content handling in a more maintainable way, without a reliance upon comparing against a magic string. Ideally we can move away from stuff like:
```php
if ($hub['Title'] == '[Theme - Mature]') {
    if ($userDetails && BitSet($userDetails['websitePrefs'], $matureContentPref)) {
        break;
    }
    $gate = true;
}
```

Configuration of this value is backed by a policy:
```php
public function toggleHasMatureContent(User $user, GameSet $gameSet): bool
{
    return $user->hasAnyRole([
        Role::ADMINISTRATOR,
        Role::DEVELOPER_STAFF, // <---- maybe this should be removed and only allow admin?
    ]);
}
```

> As it stands, this is specifically for the "mature content warning", but it seems like it could be for any type of content warning (i.e. seizure warnings).

While I agree that we should support these, mature content has different handling requirements. We wouldn't block access to games with seizure risks, but would want to warn users on those respective game pages. The nuance is likely in how "elevated" a warning type needs to be.

For future content warning types, we could use string enum columns, JSONB, or even tags. Each approach has trade-offs that will need to be evaluated, and right now I'm not sure what the best choice is.

> Maybe it should be renamed to has_mature_content or something similar. Or changed to an enum that can be set to "mature" or "seizure" or whatever other warnings we want to add in the future.

I've renamed this field to `has_mature_content` in latest.